shared/reducers/chat2.tsx

```diff
@@ -499,11 +499,18 @@ export default (_state: Types.State = initialState, action: Actions): Types.Stat
         }
         return
       case Chat2Gen.unfurlTogglePrompt: {
-        const {show, domain} = action.payload
-        draftState.unfurlPromptMap = draftState.unfurlPromptMap.updateIn(
-          [action.payload.conversationIDKey, action.payload.messageID],
-          (prompts = I.Set<string>()) => (show ? prompts.add(domain) : prompts.delete(domain))
-        )
+        const {show, domain, conversationIDKey, messageID} = action.payload
+        const unfurlPromptMap = new Map(draftState.unfurlPromptMap || [])
+        const mmap = new Map(unfurlPromptMap.get(conversationIDKey) || [])
+        const prompts = new Set(mmap.get(messageID) || [])
+        if (show) {
+          prompts.add(domain)
+        } else {
+          prompts.delete(domain)
+        }
+        mmap.set(messageID, prompts)
+        unfurlPromptMap.set(conversationIDKey, mmap)
+        draftState.unfurlPromptMap = unfurlPromptMap
```

Comment:
These are tricky. I wonder if some helper like `updateMap` could help. Maybe like
```
draftState.unfurlPromptMap = updateMap(draftState.unfurlPromptMap, (map) => {
  const mmap = updateMap(unfurlPromptMap.get(conversationIDKey) || [], (mmap) =>  {
    const prompts = new Set(mmap.get(messageID) || [])
    if (show) {
      prompts.add(domain)
    } else {
      prompts.delete(domain)
    }
    return mmap.set(messageID, prompts)
  })
  return map.set(conversationIDKey, mmap)
})
```
Where `updateMap` calls `new Map` on the first param. It's not much simpler code-wise but I think mentally it's easier to reach for `updateMap` every time you change a map in a reducer. Anyway, just an idea.

Reply:
i think its hard to get this to type correctly. hopefully a bunch of this goes away with 
`?.` and Map support in the next immer

Reply:
its also likely simpler to just not have all these nested things at all and just flatten it out. just make the keys `convoid + : + messageid`

Reply:
Yeah agreed we should try to avoid nesting these